atomic_reactor/plugins/pre_koji.py

```diff
@@ -45,7 +45,17 @@ def run(self):
             self.log.error("provided target '%s' doesn't exist", self.target)
             raise RuntimeError("Provided target '%s' doesn't exist!" % self.target)
         tag_info = self.xmlrpc.getTag(target_info['build_tag_name'])
+
+        if not tag_info or 'name' not in tag_info.keys():
+            self.log.warning("No tag info was retrieved")
+            return
+
         repo_info = self.xmlrpc.getRepo(tag_info['id'])
+
+        if not repo_info or 'id' not in repo_info.keys():
```

Comment:
Is there a case where a `getRepo` returns a dict with some keys but without the `'id'` key? I'd expect it to return either None or an empty dict, which will both be checked by the first part conditional.